scripts/calculateRevenue/protocols/somm/getVaults.ts

```diff
@@ -1,8 +1,68 @@
+import { Address, isAddress } from 'viem'
 import { NetworkId } from '../../../types'
+import { fetchWithTimeout } from '../../../utils/fetchWithTimeout'
 import { VaultInfo } from './types'
 
-// TODO(ENG-202): Dynamically fetch vaults from the Somm API
-export function getVaults(): VaultInfo[] {
+const SUPPORTED_SUFFIXES = ['arbitrum', 'optimism']
+
+export async function getVaults(): Promise<VaultInfo[]> {
+  try {
+    const result = await fetchWithTimeout('https://api.sommelier.finance/tvl')
+    if (!result.ok) {
+      throw new Error(
+        `Failed to fetch vaults from the Somm API: ${result.status} ${result.statusText}`,
+      )
+    }
+
+    const { Response: response } = await result.json()
+    return extractVaultInfo(response)
+  } catch (error) {
+    console.error('Error in getVaults:', error)
+    return getFallbackVaults()
```

Comment:
any reason to have a fallback? should we just throw and fail the script if vault fetching fails?

Reply:
Changed in 138603b24ee22bccaf82e0c27a3fc8aee7f64bb6!